packager/react-packager/src/Server/index.js

```diff
@@ -291,7 +291,8 @@ class Server {
       // Clear cached bundles in case user reloads
       this._clearBundles();
       this._hmrFileChangeListener(absPath, this._bundler.stat(absPath));
-      return;
+      // when HMR clients connected, also allow live reloading clients
+      // return;
```

Comment:
I think this is the single required change to make this work but I'm not sure what would be the best way to support this feature without affecting HMR performance.

The problem is that inside of `_hmrFileChangeListener` lots of things happen asynchronously so if you remove the `return` those async operations will compete for CPU and IO with the Live Reload changes. For Live Reload it doesn't matter if the change takes a couple more of milliseconds but for HMR it does.

Even if we turn `_hmrFileChangeListener` into a function that returns a promise and wait to execute the other listeners, we'll end up slowing the HMR codepath because the code is not injected on the client until the sourcemaps for the new file are requested and returned (see https://github.com/facebook/react-native/blob/235b16d93287061a09c4624e612b5dc4f960ce47/Libraries/Utilities/HMRClient.js#L112-L135)

For this change to not affect performance we'll need to also include the sourcemaps on the original request response.

cc @satya164